### DISCUSSION TITLE
is there an example that combines `useContractWrite` and `useWaitForTransaction`

### DISCUSSION THREAD

**Discussion by ruhaise** [wevm/wagmi#173] [Q&A]
I'm using `useContractWrite` to initialize a transaction from the frontend and I want to show the button state loading until the transaction is mined, I have read the documentation for `useWaitForTransaction` and it's a bit hard to follow how to combine `useContractWrite` with `useWaitForTransaction`. I'm now successfully using `useContractWrite` but the loading state within it only remains true until the transaction is send.

**Comment by ruhaise** [comment]
Does anybody know how to solve this?

**Reply by tmm** [reply]
Pass the `hash` or `wait` from `useContractWrite` into `useWaitForTransaction`:

```tsx
import { useContractWrite, useWaitForTransaction } from 'wagmi'

const App = () => {
  const [writeResult, write] = useContractWrite({
      addressOrName: '0x...',
      contractInterface: [],
    },
    'functionName',
  )
  const [waitResult, wait] = useWaitForTransaction({
    wait: writeResult.data?.wait
  })

  return ...
}
```

`waitResult` can be used to monitor the contract functions status.

**Reply by JohnSmithyy** [reply]
Is there an example of using useContractWrite within useWaitForTransaction onSuccess? I'm having trouble calling a second useContractWrite after the first completes.
